File: Geek Assistant (Android AIO)/Modules/General/Action/GA_Log.vb
Imports System.IO

Module GA_Log
    Private latestLogName As String
    Private latestLogPath As String
    Public Sub CreateLog()
        If Not Directory.Exists(GA) Then Directory.CreateDirectory(GA)
        If Not Directory.Exists($"{GA}\log") Then Directory.CreateDirectory($"{GA}\log")
        latestLogName = $"GA-log_{Now:(yyy.MM.dd)-hh.mm.ss}.log"
        latestLogPath = $"{GA}\log\{latestLogName}"
        File.Create(latestLogPath).Dispose()
        Dim swriter As StreamWriter = My.Computer.FileSystem.OpenTextFileWriter(latestLogPath, False)
        swriter.WriteLine(Main.log.Text)
        swriter.Close()
    End Sub
    Public Sub LogEvent(EventName As String, lines As Integer)
        LogAppendText($"// {Now:hhh:mm:ss.ff} // {EventName} //", lines)
    End Sub

    Public Sub ResetLog()
        LogEvent("Log Cleared", 3)
        CreateLog()

        Main.log.Text = GA_Ver.Run("log")

        LogAppendText($"// Previous log saved //  {latestLogName}  //", 1)
        LogEvent("Continue", 1)
    End Sub

    Public Sub LogAppendText(logText As String, lines As Integer)
        For newline As Integer = 1 To lines
            Main.log.AppendText(vbCrLf)
        Next
        Main.log.AppendText(logText)
    End Sub

    Public Sub StopNotifyIfLogSeen()
        If Main.log.Visible And (Main.ShowLog_ErrorBlink_Timer.Enabled Or Main.ShowLog_InfoBlink_Timer.Enabled) Then
            With Main
                .ShowLog_ErrorBlink_Timer.Enabled = False
                .ShowLog_InfoBlink_Timer.Enabled = False
                If My.Settings.DarkTheme Then
                    .ShowLog_Button.Icon = My.Resources.Commands_dark_24
                Else .ShowLog_Button.Icon = My.Resources.Commands_24
                End If
                '.ShowLog_Button.BackColor = Color.White
                .ProgressBarLabel.ForeColor = SystemColors.ControlText
            End With
        End If
    End Sub

    Public Sub ClearIf30days()
        If Not Directory.Exists(GA_logs) Then Exit Sub 'Exit if doesn't exist
        For Each file As FileInfo In New DirectoryInfo(GA_logs).GetFiles("*.txt")
            If (Now - file.CreationTime).Days > 30 Then file.Delete()
        Next
    End Sub
End Module

File: Geek Assistant (Android AIO)/Modules/General/Action/GA_Log.cs
﻿using System.Drawing;
using GeekAssistant.Forms;
using System.IO;
using Markdig;
using Microsoft.VisualBasic; // Install-Package Microsoft.VisualBasic
using System.Windows.Forms;

internal static partial class GA_Log {
    private static string latestLogName;
    private static string latestLogPath;

    private static Home Home = null;
    private static void RefresHome() {
        foreach (Form h in Application.OpenForms)
            if (h.GetType() == typeof(Home))
                Home = (Home)h;
    }
    public static void CreateLog() {
        RefresHome();

        if (!Directory.Exists(c.GA))
            Directory.CreateDirectory(c.GA);
        if (!Directory.Exists($@"{c.GA}\log"))
            Directory.CreateDirectory($@"{c.GA}\log");
        latestLogName = $"GA-log_{DateAndTime.Now:(yyy.MM.dd)-hh.mm.ss}.log";
        latestLogPath = $@"{c.GA}\log\{latestLogName}";
        File.Create(latestLogPath).Dispose();
        StreamWriter swriter = new StreamWriter(latestLogPath);
        swriter.WriteLine(Home.log.Text);
        swriter.Close();
    }

    public static void LogEvent(string EventName, int lines) {
        LogAppendText($"// {DateAndTime.Now:hhh:mm:ss.ff} // {EventName} //", lines);
    }

    public static void ResetLog() {
        RefresHome();

        LogEvent("Log Cleared", 3);
        CreateLog();
        Home.log.Text = GA_Ver.Run("log");
        LogAppendText($"// Previous log saved //  {latestLogName}  //", 1);
        LogEvent("Continue", 1);
    }

    public static void LogAppendText(string logText, int lines) {
        RefresHome();

        // Dim StringLines As String() = logText.Split(New String() {Environment.NewLine}, StringSplitOptions.None)

        for (int newline = 1, loopTo = lines; newline <= loopTo; newline++) {
            // Main.htmlLog.DocumentText &= br
            Home.log.Text += c.n;
        }
        Home.log.Text += logText;
        // For Each item In StringLines
        // Main.htmlLog.DocumentText &= Markdown.ToHtml(StringLines(item))
        // If StringLines.Count - 1 <> item Then Main.htmlLog.DocumentText &= br 'new line if not finished
        // Next

        // Main.htmlLog.DocumentText &= logText 'Markdown.ToHtml(logText)
    }

    public static void StopNotifyIfLogSeen() {
        RefresHome();

        if (Home.log.Visible & (Home.ShowLog_ErrorBlink_Timer.Enabled | Home.ShowLog_InfoBlink_Timer.Enabled)) {
            {
                Home.ShowLog_ErrorBlink_Timer.Stop();
                Home.ShowLog_InfoBlink_Timer.Stop();
                if (c.S.DarkTheme) Home.ShowLog_Button.Icon = prop.x24.Commands_dark_24;
                else Home.ShowLog_Button.Icon = prop.x24.Commands_24;

                // .ShowLog_Button.BackColor = Color.White
                Home.ProgressBarLabel.ForeColor = SystemColors.ControlText;
            }
        }
    }

    public static void ClearIf30days() {
        if (!Directory.Exists(c.GA_logs))
            return; // Exit if doesn't exist
        foreach (FileInfo file in new DirectoryInfo(c.GA_logs).GetFiles("*.txt")) {
            if ((DateAndTime.Now - file.CreationTime).Days > 30)
                file.Delete();
        }
    }
}